cabal-install/src/Distribution/Client/ProjectPlanning.hs

```diff
@@ -304,12 +304,14 @@ sanityCheckElaboratedPackage ElaboratedConfiguredPackage{..}
 --
 rebuildProjectConfig :: Verbosity
                      -> HttpTransport
+                     -> Bool
```

Comment:
I am a little but reluctant to add boolean params to functions (boolean blindness etcetc). 
Have you explore how could be implmented without adding the param? Afaics it is only called with `True` in one location. Could be the output of the function altered in the call site where is called with `ignoreLocalProjectFile ==true`?

In case the function should be parametized, could we use another param (`cliConfig :: ProjectConfig` f.e.?) to pass that info (without forcing its actual semantics)?

Reply:
That makes sense. I've now moved most functions over to use the passed in cli config instead of a bool. The only function I can't use the cli config is `rebuildProjectConfig` but I changed that to a `Flag Bool` and a comment explaining which flag it is for. I also changed way the "--ignore-project" flag is set to allow for `NoFlag` if neither param is set and to set the flag to false if the "--project-file" is set. This should align to how flags are defined on the `Flag` data type comments. I also added notes for how each state can be set, so it looks less confusing.

Reply:
It looks better with comment, thanks
I would prefer to not use a new Bool param at all but i dont see an easy way to avoid it so fine for me